app/models/application_record.rb

```diff
@@ -1,3 +1,9 @@
 class ApplicationRecord < ActiveRecord::Base
   self.abstract_class = true
+
+  def self.validates_lengths(max_lengths)
```

Comment:
I like the way `max_lengths` is passed in, but I’m not immediately comfortable with `ApplicationRecord.validates_lengths`.

Trying to put my thoughts as clearly as possible:

- I feel that promoting field/length mapping to a constant implies a significance field lengths don't merit, which in turn creates extra cognitive load
- using plain `validates :name, length: { maximum: 10 }` style validation is boring, predictable and easy to change
- if we wanted to add a condition to any of the validations currently handled by `validates_lengths` we would have to remove every length validation of that model from `validates_lengths`, because what is the point of a MAX_LENGTHS var without all max lengths of fields in it. This model would in turn become inconsistent with the rest of the system and `validates_lengths` would become a curiosity

Reply:
I see your point, I am not sure about having  `validates_lengths` on `ApplicationRecord` too.
I'll move it back to the model.
On the other side, we are polluting the models with many validations, so I'd prefer to group the validation for length on a constant. In the future, we might consider creating FormObject to totally extract the validations from models.